### ISSUE TITLE
QuerySelector on elements behaving like querySelectors on full document

### ISSUE THREAD

**Issue by Aulig** [dint-dev/universal_html#34]
Hi, I migrated to null-safety yesterday and in the process upgraded universal_html. I'm now encountering an issue when using querySelectors on elements (not the document). It seems like the querySelector is executed on almost the entire document instead of just the element and its children.

The regular html package produces the expected result.
```dart
import 'package:html/parser.dart';
import 'package:http/http.dart';
import 'package:universal_html/parsing.dart';
import 'package:universal_html/html.dart' as uhtml;

main() async {

	String html = (await get(Uri.parse("https://github.com/"))).body;

	print("html:");
	print(parse(html).querySelectorAll("a").length);
	print(parse(html).querySelector("nav").querySelectorAll("a").length);

	print("universal_html:");
	print(parseHtmlDocument(html).querySelectorAll("a").length);
	print(parseHtmlDocument(html).querySelector("nav").querySelectorAll("a").length);

	// workaround
	print(uhtml.Element.html(parseHtmlDocument(html).querySelector("nav").outerHtml).querySelectorAll("a").length);
}
```
Output:
```
html:
119
29
universal_html:
119
114
Removing disallowed element ...
29
```

Am I doing something wrong or is this an intended change? I'd rather not have to use the workaround as it seems like a performance issue. The other alternative would be changing my code to chain the selectors like "nav a" where html and universal_html provide the same result. (Not a 1:1 replacement though as there can be multiple nav elements)

**Comment by Aulig** [comment]
It looks like the issue was introduced in the upgrade from 1.2.4 to 2.0.0. Before that, html and universal_html behave the same in this example.

**Comment by terrier989** [comment]
Thanks for the issue report! It has been fixed in v2.0.5.

**Comment by Aulig** [comment]
Looks good, thanks for your effort!
